File: hw6/Rectangle.cpp
#include    "Rectangle.h"

Rectangle::Rectangle(const string &color, int inp_h, int inp_w):Shape(color){
    height = inp_h;
    width = inp_w;
    cout << "Rectangle construct." << endl;
}

Rectangle::~Rectangle(){
    cout << "Rectangle destruct." << endl;
}

void Rectangle::print(){
    double area = 0;
    string color;
    area = get_area();
    color = getcolor();
    cout << color << " rectangle, height " << height << ", width " << width << ", area " << area << endl;  //red rectangle, height 8, width 6, area 48
}

double Rectangle::get_area(){
    double area = 0;
    area = height * width;
    return area;
}

File: hw6/Rectangle.java
import hw6.Shape;

public class Rectangle extends Shape{
    private int height;
    private int width;
    public Rectangle(String color, int h, int w){
        super(color);
        height = h;
        width = w;
    }
    public void print(){
        double area = 0;
        String color;
        area = get_area();
        color = get_color();
        System.out.println(color + " rectangle, height " + height + ", width " + width + ", area " + area);
    }
    public double get_area(){
        double area = 0;
        area = height * width;
        return area;
    }
}